sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -336,9 +336,11 @@ private[sql] object DataSourceScanExec {
   def create(
       output: Seq[Attribute],
       rdd: RDD[InternalRow],
-      relation: BaseRelation,
+      logicalRelation: LogicalRelation,
       metadata: Map[String, String] = Map.empty,
       metastoreTableIdentifier: Option[TableIdentifier] = None): DataSourceScanExec = {
+    val relation: BaseRelation = logicalRelation.relation
+
     val outputPartitioning = {
       val bucketSpec = relation match {
```

Comment:
Do you mean?
```scala
def toAttribute(colName: String): Attribute = output.find(_.name == colName).get

bucketSpec.map { spec =>
  val numBuckets = spec.numBuckets
  if (spec.bucketColumnNames.forall(colName => output.exists(_.name == colName))) {
    val bucketColumns = spec.bucketColumnNames.map(toAttribute)
    HashPartitioning(bucketColumns, numBuckets)
  } else {
    UnknownPartitioning(0)
  }
}.getOrElse {
  UnknownPartitioning(0)
}
```

But the sql below also can't be benefited from bucketing but it's satisfy all conditions(the conf is enabled, and the relation is bucketed, and the output contains all bucketing columns).
```sql
SELECT i, j, k FROM bucketed_table;
```
So we need to use `groupbyColums` or `joinColums` instead of `output`?